### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by CesarHerr**
- Remove all hardcoded items from the tasks array. ✔️ 
- Create a new JavaScript file for the new functionality. ✔️ 
- Implement a function for adding a new task (add a new element to the array). ✔️ 
- Implement a function for deleting a task (remove an element from the array). ✔️ 
- Implement a function for editing task descriptions. ✔️ 
- By default new tasks should have the property completed set to false and the property index set to the value of the new array 
   length. ✔️ 
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. ✔️ 
- All changes to the To Do List should be saved in local storage. ✔️ 

image

### CODE REVIEW

File: modules/crud.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    this.displayList();
+    this.enterEvent();
+  }
+
+  add(text) {
+    const newTodo = new Todo(text);
+    this.list.push(newTodo);
+    localStorage.setItem('list', JSON.stringify(this.list));
+    this.displayList();
+  }
```
Comment (Reem-lab) [inline modules/crud.js:17]: - [ ] Great job working on all the functionality but you still have one thing to change when i see the tasks on localstorage I can't see that it has indexes so whenever the user remove the task the indexes should be updated as required ✅⬇🔥🤩 so if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1) so kindly implement the indexes here to each task 🔥💯


image


- I can see just completed and the name of the task
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @CesarHerr, 👋🏻👋🏻

Great job so far! 🤩💯✅
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- No Linters error 💯🤩
- Followed GitHub Flow 💯🤩
- Great job working on the functionality 💯
- Professional README ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/crud.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  constructor() {
+    this.list = JSON.parse(localStorage.getItem('list')) || [];
+    this.displayList();
+    this.enterEvent();
+    this.indexes();
+  }
+
+  add(text) {
+    const newTodo = new Todo(text);
+    newTodo.index = this.list.length;
```
Comment (Davidosky007) [inline modules/crud.js:15]: * As the previous CR stated your index if you're adding a 5th task to the list, the index of that task should equal 5 for example first todo == index 1, fourth todo == index 4, kindly fix this same thing also should be done when deleting if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @CesarHerr 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No Linters error 💯🤩
* Followed GitHub Flow 💯🤩
* Great job working on the functionality 💯
* Professional README ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmiiorji — APPROVED** [review]
Hi @CesarHerr, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<div align='center'>

</div>

## Good Highlights✔️

- Requested changes have been implemented
- Application works as expected


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**